avocado/plugins/yaml_to_mux.py

```diff
@@ -0,0 +1,256 @@
+    if len(path) == 1:
+        path = __RE_FILE_SUBS.sub(':', path[0])
+        using = ["run"]
+    else:
+        nodes = __RE_FILE_SUBS.sub(':', path[0]).strip('/').split('/')
+        using = [node for node in nodes if node]
+        if not path[0].startswith('/'):  # relative path, put into /run
+            using.insert(0, 'run')
+        path = __RE_FILE_SUBS.sub(':', path[1])
+
+    # Load the tree
+    with open(path) as stream:
+        loaded_tree = yaml.load(stream, Loader)
+        loaded_tree = tree_node_from_values('', loaded_tree)
+
+    # Add prefix
+    if using:
+        loaded_tree.name = using.pop()
+        while True:
+            if not using:
+                break
+            loaded_tree = cls_node(using.pop(), children=[loaded_tree])
+        loaded_tree = cls_node('', children=[loaded_tree])
+    return loaded_tree
+
+
+def create_from_yaml(paths, debug=False):
+    """
+    Create tree structure from yaml-like file
+    :param fileobj: File object to be processed
+    :raise SyntaxError: When yaml-file is corrupted
+    :return: Root of the created tree structure
+    """
+    def _merge(data, path):
+        """ Normal run """
+        data.merge(_create_from_yaml(path))
+
+    def _merge_debug(data, path):
+        """ Use NamedTreeNodeDebug magic """
+        node_cls = tree.get_named_tree_cls(path)
+        data.merge(_create_from_yaml(path, node_cls))
+
+    if not debug:
+        data = tree.TreeNode()
+        merge = _merge
+    else:
+        data = tree.TreeNodeDebug()
+        merge = _merge_debug
+
+    path = None
+    try:
+        for path in paths:
+            merge(data, path)
+    # Yaml can raise IndexError on some files
+    except (yaml.YAMLError, IndexError) as details:
+        if 'mapping values are not allowed in this context' in str(details):
+            details = ("%s\nMake sure !tags and colons are separated by a "
+                       "space (eg. !include :)" % details)
+        msg = "Invalid multiplex file '%s': %s" % (path, details)
+        raise IOError(2, msg, path)
+    return data
+
+
+class YamlToMux(CLI):
+
+    """
+    Registers callback to inject params from yaml file to the
+    """
+
+    name = 'yaml_to_mux'
+    description = "YamlToMux options for the 'run' subcommand"
+
+    def configure(self, parser):
+        """
+        Configures "run" and "multiplex" subparsers
+        """
+        if not MULTIPLEX_CAPABLE:
+            return
+        for name in ("run", "multiplex"):
+            subparser = parser.subcommands.choices.get(name, None)
+            if subparser is None:
+                continue
+            mux = subparser.add_argument_group("yaml to mux options")
+            mux.add_argument("-m", "--multiplex", nargs='*', dest="multiplex",
+                             default=None, metavar="FILE",
+                             help="Location of one or more Avocado multiplex "
+                             "(.yaml) FILE(s) (order dependent)")
```

Comment:
I'd say "order is significant" (or even omit with the reasoning that this is really a short message and users should be aware of how the multiplexer works by consulting the more complete docs).